x-pack/plugins/security_solution/public/timelines/components/timeline/query_tab_content/query_tab_unified_components.test.tsx

```diff
@@ -0,0 +1,615 @@
+}));
+
+jest.mock('../../../../common/lib/kuery');
+
+jest.mock('../../../../common/hooks/use_experimental_features');
+
+const useIsExperimentalFeatureEnabledMock = jest.fn((feature: keyof ExperimentalFeatures) => {
+  if (feature === 'unifiedComponentsInTimelineEnabled') {
+    return true;
+  }
+  return allowedExperimentalValues[feature];
+});
+
+jest.mock('../../../../common/lib/kibana');
+
+// unified-field-list is is reporiting multiple analytics events
+jest.mock(`@kbn/analytics-client`);
+
+const TestComponent = (props: Partial<ComponentProps<typeof QueryTabContent>>) => {
+  const testComponentDefaultProps: ComponentProps<typeof QueryTabContent> = {
+    timelineId: TimelineId.test,
+    renderCellValue: DefaultCellRenderer,
+    rowRenderers: defaultRowRenderers,
+  };
+
+  const dispatch = useDispatch();
+
+  // populating timeline so that it is not blank
+  useEffect(() => {
+    dispatch(
+      timelineActions.applyKqlFilterQuery({
+        id: TimelineId.test,
+        filterQuery: {
+          kuery: {
+            kind: 'kuery',
+            expression: '*',
+          },
+          serializedQuery: '*',
+        },
+      })
+    );
+  }, [dispatch]);
+
+  return <QueryTabContent {...testComponentDefaultProps} {...props} />;
+};
+
+const renderTestComponents = (props?: Partial<ComponentProps<typeof TestComponent>>) => {
+  return render(<TestComponent {...props} />, {
+    wrapper: TestProviders,
+  });
+};
+
+const loadPageMock = jest.fn();
+
+const useTimelineEventsMock = jest.fn(() => [
+  false,
+  {
+    events: cloneDeep(mockTimelineData),
+    pageInfo: {
+      activePage: 0,
+      totalPages: 10,
+    },
+    refreshedAt: Date.now(),
+    totalCount: 70,
+    loadPage: loadPageMock,
+  },
+]);
+
+describe('query tab with unified timeline', () => {
+  const kibanaServiceMock: StartServices = createStartServicesMock();
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  beforeEach(() => {
+    const ONE_SECOND = 1000;
+    jest.setTimeout(10 * ONE_SECOND);
+    HTMLElement.prototype.getBoundingClientRect = jest.fn(() => {
+      return {
+        width: 1000,
+        height: 1000,
+        x: 0,
+        y: 0,
+      } as DOMRect;
+    });
+
+    (useKibana as jest.Mock).mockImplementation(() => {
+      return {
+        services: kibanaServiceMock,
+      };
+    });
+
+    (useTimelineEvents as jest.Mock).mockImplementation(useTimelineEventsMock);
+
+    (useTimelineEventsDetails as jest.Mock).mockImplementation(() => [false, {}]);
+
+    (useSourcererDataView as jest.Mock).mockImplementation(() => ({
+      ...mockSourcererScope,
+    }));
+
+    (useIsExperimentalFeatureEnabled as jest.Mock).mockImplementation(
+      useIsExperimentalFeatureEnabledMock
+    );
+  });
+
+  describe('render', () => {
+    it('should render unifiedDataTable in timeline', async () => {
+      renderTestComponents();
+      await waitFor(() => {
+        expect(screen.getByTestId('discoverDocTable')).toBeVisible();
```

Comment:
Should we update the testId for our implementation to be able to better track telemetry? Can we use the same test id that was used for the timeline previously?

Reply:
Actually, we do have our own test ID. but here want to check if the actual unified component has completely rendered and actually table is available for assertion. 

For telemetry, we can surely use our own testIDs.

Does this explanation makes sense?